src/backend/renderer.js

```diff
@@ -743,8 +753,24 @@ export function attach(
       enqueueMount(fiber, parentFiber);
     }
 
-    if (fiber.child !== null) {
-      mountFiber(fiber.child, shouldEnqueueMount ? fiber : parentFiber);
+    const isTimedOutSuspense =
+      fiber.tag === ReactTypeOfWork.SuspenseComponent &&
+      fiber.memoizedState !== null;
+
+    if (isTimedOutSuspense) {
+      // Special case: if Suspense mounts in a timed-out state,
+      // get the fallback child from the inner fragment and mount
+      // it as if it was our own child. Updates handle this too.
+      const primaryChildFragment = fiber.child;
+      const fallbackChildFragment = primaryChildFragment.sibling;
+      const fallbackChild = fallbackChildFragment.child;
```

Comment:
This part looks correct. It corresponds to this part of `updateSuspenseComponent`:

https://github.com/facebook/react/blob/49595e921dcb748c0422bc7aaadf2d3288ff1c46/packages/react-reconciler/src/ReactFiberBeginWork.js#L1434-L1465